lib/workers/bin/provider_worker

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env ruby
+
+require File.expand_path("../../../config/environment", __dir__)
+
+ems_id = ENV["EMS_ID"]
+
+ems      = ExtManagementSystem.find(ems_id)
+all_emss = [ems] + ems.child_managers
+ems_opts = all_emss.map do |e|
+  e.attributes.merge(
+    "endpoints"       => e.endpoints,
+    "authentications" => e.authentications
+  )
+end
+
+provider_opts = {
+  :ems       => ems_opts,
+  :messaging => MiqQueue.messaging_client_options,
+  :settings  => Settings.to_h
+}
+
+# Create a temp file and immediately unlink it to create a
+# secure hidden file to be used for the child process' stdin.
+stdin_tmp = Tempfile.new
+File.unlink(stdin_tmp.path)
```

Comment:
I recall there being a more secure way to do this, I think with an flock?  I'll have to search around.

Reply:
`flock` is advisory not mandatory so I doubt that was it.

This works because `mkstemp` filenames cannot be guessed ahead of time thus the file can't be open prior to creation.  `mkstemp` also opens with `O_EXCL` to make sure the open will fail if the file exists even if they were somehow to guess the tempfile name.  The only tweak to this code that I've found has involved creating the tempfile in a non-shared directory only accessible by the file owner's user (like `/tmp/users/#{userid}`) which we could certainly do

Reply:
Might be worth doing that - security stuff gets me paranoid :)

Reply:
@Fryguy this looks promising actually https://www.freedesktop.org/software/systemd/man/systemd.exec.html#PrivateTmp= I'm going to go this route since it would (theoretically) completely mitigate any file access from outside the service